test/integration/roles/test_ec2_key/tasks/main.yml

```diff
@@ -249,7 +249,7 @@
     name='{{ec2_key_name}}'
     state=present
     key_material='{{ item }}'
-  with_file: sshkey ~ '.pub'
+  with_file: '{{sshkey}}.pub'
```

Comment:
sshkey is the location of the private key file which is created in the setup role.  {{sshkey}}.pub then becomes the public key which should be uploaded to AWS.

I have no idea if the original statement ever worked in some older version of Ansible.  I doubt it.